### DISCUSSION TITLE
Does GPT Researcher have an API ?

### DISCUSSION THREAD

**Discussion by djuillard** [assafelovic/gpt-researcher#715] [Q&A]
I am wondering whether it is possible to use the service via API (even if we have to self-host the instance) in order to automate researchs and skip the front end part ?
Thanks

**Comment by danieldekay** [comment]
There is https://docs.gptr.dev/docs/gpt-researcher/pip-package with a FastAPI example.

**Reply by djuillard** [reply]
Excellent ! Thanks !!!

**Comment by Cloudore** [comment]
Is the api deprecated?

**Reply by ElishaKay** [reply]
Welcome @Cloudore 

You're welcome to deploy your own GPTR API:

See the docs on [Linux Deployment](https://docs.gptr.dev/docs/gpt-researcher/getting-started/linux-deployment) - we might make this API easier to consume in the near future
